ru/turing/courses/lesson3/Registry.java

```diff
@@ -1,13 +1,56 @@
 package ru.turing.courses.lesson3;
 
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
 
-//todo Задание со звездочкой - сделать Registry через generics.
-// Для этого почитать, что такое дженерики и о том, как заставить класс с ними работать
-// Сделать абстрактный доменный класс (похожий на Animal из прошлого урока для этого)
-public class Registry {
-    private final Map<Object, Object> storage = new HashMap<>(); //todo поменять Object, Object на то, что будет у вашего класса в примере
+public class Registry<T extends Number> {
+    private final Map<Integer, T> storage = new HashMap<>();
 
-    //todo написать и реализовать методы
+
+    public void add(Integer key, T value) {
+        this.storage.put(key, value);
+    }
+
+
+    public T removeByKey(Integer key) {
+        T value = this.storage.get(key);
+        this.storage.remove(key);
+        return value;
+    }
+
+
+    public ArrayList removeByValue(T value) {
```

Comment:
```suggestion
    public List<T> removeByValue(T value) {
```
Возвращаем лист. Какая реализация именно -- это уже дело второе, затрагивает только внутреннюю работу метода